libs/api/domains/islykill/src/lib/islykill.service.ts

```diff
@@ -0,0 +1,139 @@
+import { User } from '@island.is/auth-nest-tools'
+
+import { IslykillSettings, IslykillUpdateResponse } from './islykill.type'
+
+@Injectable()
+export class IslykillService {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private readonly islyklarApi: IslyklarApi,
+  ) {}
+
+  async updateIslykillSettings(
+    nationalId: User['nationalId'],
+    { email, mobile }: { email: string; mobile?: string },
+  ): Promise<IslykillUpdateResponse> {
+    const inputUserData: PublicUser = {
+      ssn: nationalId,
+      email,
+      mobile,
+    }
+    const errorMsg = 'Unable to update islykill settings for user'
+    const apiData = await this.islyklarApi
+      .islyklarPut({
+        user: inputUserData,
+      })
+      .then(() => {
+        return {
+          nationalId,
+          valid: true,
+        }
+      })
+      .catch((e) => {
+        this.logger.error(errorMsg, {
+          category: 'islykill-settings',
+          nationalId,
+          exception: e,
+        })
+        throw new BadRequestException(e, errorMsg)
+      })
+    return apiData
+  }
+
+  async getIslykillSettings(
+    nationalId: User['nationalId'],
+  ): Promise<IslykillSettings> {
+    const errorMsg = 'Unable to lookup islykill settings for user'
+    const apiData = await this.islyklarApi
+      .islyklarGet({
+        ssn: nationalId,
+      })
+      .then((userData: PublicUser) => {
+        return {
+          nationalId,
+          email: userData.email,
+          mobile: userData.mobile,
+          bankInfo: userData.bankInfo,
+          lastLogin: userData.lastLogin,
+          nextLastLogin: userData.nextLastLogin,
+          lastPassChange: userData.lastPassChange,
+          canNudge: userData.canNudge,
+          onlyCert: userData.onlyCert,
+          nudgeLastAsked: userData.nudgeLastAsked,
+        }
+      })
+      .catch((e) => {
+        if (e.status === 404) {
+          return {
+            nationalId,
+            noUserFound: true,
+          }
+        }
+        this.logger.error(errorMsg, {
+          category: 'islykill-settings',
+          nationalId,
+          exception: e,
+        })
+        throw new BadRequestException(e, errorMsg)
+      })
+    return apiData
+  }
+
+  async createIslykillSettings(
+    nationalId: User['nationalId'],
+    { email, mobile }: { email: string; mobile?: string },
+  ) {
+    const inputUserData: PublicUser = {
+      ssn: nationalId,
+      email,
+      mobile,
+    }
+
+    const errorMsg = 'Unable to create islykill settings for user'
+    const apiData = await this.islyklarApi
+      .islyklarPost({
+        user: inputUserData,
+      })
+      .then(() => {
+        return {
+          nationalId,
+          valid: true,
+        }
+      })
+      .catch((e) => {
+        this.logger.error(errorMsg, {
+          category: 'islykill-settings',
+          nationalId,
+          exception: e,
+        })
+        throw new BadRequestException(e, errorMsg)
```

Comment:
Fyi. the enhanced fetch already logs request issues pretty well. And we should never log national ids (they're masked anyways).

Maybe a friendly BadRequestException is enough?